relational_operators/TextScanOperator.cpp

```diff
@@ -274,439 +116,293 @@ TextScanWorkOrder::TextScanWorkOrder(const std::size_t query_id,
 
 void TextScanWorkOrder::execute() {
   const CatalogRelationSchema &relation = output_destination_->getRelation();
+  std::vector<Tuple> tuples;
 
-  string current_row_string;
-  if (is_file_) {
-    FILE *file = std::fopen(filename_.c_str(), "r");
-    if (file == nullptr) {
-      throw TextScanReadError(filename_);
-    }
+  constexpr std::size_t kSmallBufferSize = 0x4000;
```

Comment:
From experiments this is the best size for both _slowdisk_ and _fastdisk_ on the cloudlab machines. The observation is that _slowdisk_ loading favors smaller (256K) segment size and _fastdisk_ loading favors larger (1M~4M, but 256K is okay) segment size. I haven't dived into the underlying details though.